pkg/trace/api/api.go

```diff
@@ -762,3 +762,23 @@ func getMediaType(req *http.Request) string {
 	}
 	return mt
 }
+
+// getDirector returns a director for a httputil.ReverseProxy. It enriches requests with headers setting custom
+// container and additional tags and overwrites the user agent header.
+func headerDecoratorDirector(tags string, metricSuffix string) func(req *http.Request) {
+	return func(req *http.Request) {
+		req.Header.Set("Via", fmt.Sprintf("trace-agent %s", info.Version))
+		if _, ok := req.Header["User-Agent"]; !ok {
+			// explicitly disable User-Agent so it's not set to the default value
+			// that net/http gives it: Go-http-client/1.1
+			// See https://codereview.appspot.com/7532043
+			req.Header.Set("User-Agent", "")
+		}
+		containerID := req.Header.Get(headerContainerID)
+		if ctags := getContainerTags(containerID); ctags != "" {
+			req.Header.Set("X-Datadog-Container-Tags", ctags)
+		}
+		req.Header.Set("X-Datadog-Additional-Tags", tags)
+		metrics.Count("datadog.trace_agent."+metricSuffix, 1, nil, 1)
```

Comment:
I think you can remove this along with the function argument. This is a duplicate of “proxy_request” metric. At this point we are barely setting headers so it’s unfair to say that any request happened. In profiling, you can move the metric to the RoundTrip method. 

I also think it’s slightly odd that this func is in a file where it’s not used but I understand the reasoning and don’t feel strongly about it.

Reply:
Yeah, I just noticed that as well. Pushed a fix already.